File: test57_out/test57/SetFuncRefOnHost/SetFuncRefOnHost.c

/*
 * ======== string literals ========
 */

static const char test57_SetFuncRefOnHost__s__fileName[] = "SetFuncRefOnHost.p";
static const char test57_SetFuncRefOnHost__s__0[] = "SetFuncRefOnHost";

/*
 * ======== code injections ========
 */

 #include <stdio.h> 

/*
 * ======== function definitions ========
 */

void test57_SetFuncRefOnHost_setTheFuncRefOnTarg1__I( test57_SetFuncRefOnHost_HandlerP_handler h ) {
    (test57_SetFuncRefOnHost_myHandler3_Host__V) = (h);
}

void test57_SetFuncRefOnHost_setTheFuncRefOnTarg2__I( test57_SetFuncRefOnHost_HandlerP_handler h ) {
    (test57_SetFuncRefOnHost_myHandler4_Host__V) = (h);
}

void test57_SetFuncRefOnHost_setTheFuncRefOnTarg3__I( test57_SetFuncRefOnHost_HandlerP_handler h ) {
    (test57_SetFuncRefOnHost_myHandler1_Targ__V) = (h);
}

void test57_SetFuncRefOnHost_setTheFuncRefOnTarg4__I( test57_SetFuncRefOnHost_HandlerP_handler h ) {
    (test57_SetFuncRefOnHost_myHandler2_Targ__V) = (h);
}

void test57_SetFuncRefOnHost_targetTheHandler__I() {
     printf("target Handler called\n") ;

}

void test57_SetFuncRefOnHost_pollen__shutdown__E( uint8 id ) {
}

void test57_SetFuncRefOnHost_pollen__run__E() {
     printf("check myHandler1_Host\n") ;

    if ((test57_SetFuncRefOnHost_myHandler1_Host__V) != null) {
        test57_SetFuncRefOnHost_myHandler1_Host__V();
    }
     printf("check myHandler2_Host\n") ;

    if ((test57_SetFuncRefOnHost_myHandler2_Host__V) != null) {
        test57_SetFuncRefOnHost_myHandler2_Host__V();
    }
     printf("check myHandler3_Host\n") ;

    if ((test57_SetFuncRefOnHost_myHandler3_Host__V) != null) {
        test57_SetFuncRefOnHost_myHandler3_Host__V();
    }
     printf("check myHandler4_Host\n") ;

    if ((test57_SetFuncRefOnHost_myHandler4_Host__V) != null) {
        test57_SetFuncRefOnHost_myHandler4_Host__V();
    }
     printf("check myHandler1_Targ\n") ;

    if ((test57_SetFuncRefOnHost_myHandler1_Targ__V) != null) {
        test57_SetFuncRefOnHost_myHandler1_Targ__V();
    }
     printf("check myHandler2_Targ\n") ;

    if ((test57_SetFuncRefOnHost_myHandler2_Targ__V) != null) {
        test57_SetFuncRefOnHost_myHandler2_Targ__V();
    }
     printf("check myHandler3_Targ\n") ;

    if ((test57_SetFuncRefOnHost_myHandler3_Targ__V) != null) {
        test57_SetFuncRefOnHost_myHandler3_Targ__V();
    }
     printf("check myHandler4_Targ\n") ;

    if ((test57_SetFuncRefOnHost_myHandler4_Targ__V) != null) {
        test57_SetFuncRefOnHost_myHandler4_Targ__V();
    }
     exit(0) ;

}

void test57_SetFuncRefOnHost_targetInit__I() {
    test57_SetFuncRefOnHost_setTheFuncRefOnTarg2__I((test57_SetFuncRefOnHost_targetTheHandler__I));
    test57_SetFuncRefOnHost_setTheFuncRefOnTarg4__I((test57_SetFuncRefOnHost_targetTheHandler__I));
}

File: test57_out/test57/SetFuncRefOnHost/SetFuncRefOnHost.js

/*
 * ======== MODULE test57.SetFuncRefOnHost ========
 */

var $$c = function() {
    const SetFuncRefOnHost = {};
    SetFuncRefOnHost.SetFuncRefOnHost = SetFuncRefOnHost
    SetFuncRefOnHost.$name = 'test57.SetFuncRefOnHost';
    SetFuncRefOnHost.pollen$used = 0;
    SetFuncRefOnHost.ClsFuncRef = $units['test57.ClsFuncRef'];
    SetFuncRefOnHost.targetInit = new $$Ref('test57_SetFuncRefOnHost_targetInit__I');
    SetFuncRefOnHost.setTheFuncRefOnTarg1 = new $$Ref('test57_SetFuncRefOnHost_setTheFuncRefOnTarg1__I');
    SetFuncRefOnHost.setTheFuncRefOnTarg2 = new $$Ref('test57_SetFuncRefOnHost_setTheFuncRefOnTarg2__I');
    SetFuncRefOnHost.setTheFuncRefOnTarg3 = new $$Ref('test57_SetFuncRefOnHost_setTheFuncRefOnTarg3__I');
    SetFuncRefOnHost.setTheFuncRefOnTarg4 = new $$Ref('test57_SetFuncRefOnHost_setTheFuncRefOnTarg4__I');
    SetFuncRefOnHost.pollen__run = new $$Ref('test57_SetFuncRefOnHost_pollen__run__E');
    SetFuncRefOnHost.targetTheHandler = new $$Ref('test57_SetFuncRefOnHost_targetTheHandler__I');
    SetFuncRefOnHost.pollen__shutdown = new $$Ref('test57_SetFuncRefOnHost_pollen__shutdown__E');
    SetFuncRefOnHost.$$hostInit = function() {
        var $$text = '';
        SetFuncRefOnHost.clsFR = new $units['test57.ClsFuncRef'].ClsFuncRef('test57_SetFuncRefOnHost_clsFR__V'); SetFuncRefOnHost.clsFR.new_host();
        SetFuncRefOnHost.setTheFuncRefOnHost1((SetFuncRefOnHost.hostTheHandler));
        SetFuncRefOnHost.setTheFuncRefOnHost2((SetFuncRefOnHost.targetTheHandler));
        SetFuncRefOnHost.setTheFuncRefOnHost3((SetFuncRefOnHost.hostTheHandler));
        SetFuncRefOnHost.setTheFuncRefOnHost4((SetFuncRefOnHost.targetTheHandler));
        return $$text;
    }
    SetFuncRefOnHost.setTheFuncRefOnHost1 = function( h ) {
        var $$text = '';
        (SetFuncRefOnHost.myHandler1_Host) = (h);
        return $$text;
    }
    SetFuncRefOnHost.setTheFuncRefOnHost2 = function( h ) {
        var $$text = '';
        (SetFuncRefOnHost.myHandler2_Host) = (h);
        return $$text;
    }
    SetFuncRefOnHost.setTheFuncRefOnHost3 = function( h ) {
        var $$text = '';
        (SetFuncRefOnHost.myHandler3_Targ) = (h);
        return $$text;
    }
    SetFuncRefOnHost.setTheFuncRefOnHost4 = function( h ) {
        var $$text = '';
        (SetFuncRefOnHost.myHandler4_Targ) = (h);
        return $$text;
    }
    SetFuncRefOnHost.hostTheHandler = function() {
        var $$text = '';
         printf("host Handler called\n") ;

        return $$text;
    }
    SetFuncRefOnHost.pollen__uses__ = function() {
        $$bind($units['test57.ClsFuncRef'], 'pollen$used', true);
    }
    SetFuncRefOnHost.$$privateInit = function() {
        SetFuncRefOnHost.flag = true;
        SetFuncRefOnHost.flag2 = true;
    }
    return SetFuncRefOnHost;
}
var $$u = $$c();
$units['test57.SetFuncRefOnHost'] = $$u;
$units.push($$u);

